src/main/scala/codacy/plugins/test/ITest.scala

```diff
@@ -3,59 +3,143 @@ package codacy.plugins.test
 import java.io.File
 import java.nio.file.Path
 
-import codacy.plugins.docker.{DockerPlugin, Pattern, Result}
 import codacy.utils.Printer
+import com.codacy.analysis.core
+import com.codacy.analysis.core.model.{FileError, Issue, Pattern, ToolResult}
+import com.codacy.plugins.api._
+import com.codacy.plugins.api.languages.{Language, Languages}
+import com.codacy.plugins.results.traits.DockerTool
+import com.codacy.plugins.utils.PluginHelper
+
+import scala.util.{Failure, Success, Try}
+
+final case class DockerImage(name: String, version: String) {
+  override def toString: String = {
+    s"$name:$version"
+  }
+}
 
 trait ITest {
   val opt: String
 
-  def run(plugin: DockerPlugin, testSources: Seq[Path], dockerImageName: String, optArgs: Seq[String]): Boolean
+  def run(testSources: Seq[Path], dockerImage: DockerImage, optArgs: Seq[String]): Boolean
 
-  protected def filterResults(sourcePath: Path, files: Seq[File], patterns: Seq[Pattern], results: Seq[Result]): Seq[Result] = {
-    val receivedResultsTotal = results.length
 
-    if (results.nonEmpty) {
-      Printer.green(s"$receivedResultsTotal results received.")
-    } else {
-      Printer.red("No results received!")
+  protected def findLanguages(testSources: Seq[Path], dockerImage: DockerImage): Set[Language] = {
```

Comment:
While I think I understand what's going on here, I'd like to see names for the `.orElse` and the `.getOrElse` input blocks. I suggest splitting them in (inner or private) functions.

Reply:
👍 done, with lazy vals